tailnet/coordinator.go

```diff
@@ -159,8 +158,30 @@ type coordinator struct {
 	agentNameCache *lru.Cache[uuid.UUID, string]
 }
 
+func NewCore(logger slog.Logger) *Core {
```

Comment:
Thoughts on putting the coordinator in it's own package? The name `NewCore` and `Core` doesn't make much sense in the context of `tailnet` itself for providing connections.

An alternative would be putting the in-memory coordinator in it's own package.

Reply:
Makes sense to me!